funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+              <i class="fa fa-close loader js-abort"></i>
+            {{/checked_in}}
             </li>
             </ul>
           </td>
         </tr>
-      {%- endfor %}
+      {{/each}}
+      </script>
+      {% endraw %}
     </tbody>
   </table>
 {% endblock %}
 {% block footerscripts %}
 <script>
+  if (Modernizr.localstorage) {
+    window.Store = {};
+
+    //Reads from LocalStorage (key is 'checkin-queue' or 'cancelcheckin-queue')
+    window.Store.readFromLocalstorage = function(key) {
+      return JSON.parse(localStorage.getItem(key));
+    }
+    //Adds to LocalStorage (key is 'checkin-queue' or 'cancelcheckin-queue')
+    window.Store.addToLocalstorage = function(key, value) {
+      return localStorage.setItem(key, JSON.stringify(value))
+    }
+
+    window.Queue = {};
+
+    //Adds a participant_id to queue
+    window.Queue.enqueue = function(queueName, participant_id) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      if(participantList) {
+        participantList.push(participant_id);
+      }
+      else {
+        participantList = [];
+        participantList.push(participant_id);
+      }
+      return window.Store.addToLocalstorage(queueName, participantList)
+    }
+
+    //Reads first item from queue
+    //Returns undefined when queue is empty or not defined
+    window.Queue.peek = function(queueName) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      if(participantList && participantList.length) {
+        return participantList.shift();
+      }
+    }
+
+    //Reads and returns all items from queue
+    //Returns undefined when queue is empty or not defined
+    window.Queue.peekAllItems = function(queueName) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      if(participantList && participantList.length) {
+        return participantList;
+      }
+    }
+
+    //Removes item from queue and returns true
+    //Returns undefined when item not present in queue
+    window.Queue.dequeue = function(queueName, participant_id) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      var index = participantList ? participantList.indexOf(participant_id) : -1;
+      if(index !== -1) {
+        //Remove item from queue and add updated queue to localStorage
+        participantList.splice(index, 1);
+        window.Store.addToLocalstorage(queueName, participantList);
+        return true;
+      }
+    }
+  }
+
   $(document).ready(function() {
+    // To disable debug mode
+    Ractive.DEBUG = false;
+
+    var participantTableContent = new Ractive({
+      el: '#participants-table-content',
+      template: '#participant-row',
+      data: {
+        participants: ''
+      }
+    });
+
+    //updateQueue: If participant in "checkin-queue" has already been checked-in then it is removed from checkin queue
+    function updateQueue(queueName) {
```

Comment:
This function helps in avoiding sending redundant requests to server and also prevent from showing loading indicator against incorrect check-in status when UI is refreshed every 1 minute.

Example: When there are two or more check-in counters at the event and two counters does check-in for the same participant. The participant will be added to the check-in queue of both counters. Assume, the request from the first counter reaches the server first and the participant is checked-in.
When UI is refreshed in second counter's browser, this participant status will show Checked-in. Since this participant is in second counter's check-in queue too, the loading and abort indicator will be displayed which is incorrect.
So, this function will check and remove the participant from the check-in queue of second counter.